api/src/main/java/org/pmiops/workbench/workspaces/resources/WorkspaceResourceMapper.java

```diff
@@ -0,0 +1,99 @@
+import org.pmiops.workbench.conceptset.mapper.ConceptSetMapper;
+import org.pmiops.workbench.dataset.mapper.DataSetMapper;
+import org.pmiops.workbench.db.model.DbCohort;
+import org.pmiops.workbench.db.model.DbDataset;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.model.CohortReview;
+import org.pmiops.workbench.model.ConceptSet;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.model.WorkspaceResource;
+import org.pmiops.workbench.utils.mappers.CommonMappers;
+import org.pmiops.workbench.utils.mappers.MapStructConfig;
+
+@Mapper(
+    config = MapStructConfig.class,
+    collectionMappingStrategy = CollectionMappingStrategy.TARGET_IMMUTABLE,
+    uses = {
+      CohortMapper.class,
+      CohortReviewMapper.class,
+      CommonMappers.class,
+      ConceptSetMapper.class,
+      DataSetMapper.class,
+    })
+public interface WorkspaceResourceMapper {
+
+  @Mapping(target = "workspaceId", source = "dbWorkspace.workspaceId")
+  @Mapping(target = "workspaceFirecloudName", source = "dbWorkspace.firecloudName")
+  @Mapping(target = "workspaceBillingStatus", source = "dbWorkspace.billingStatus")
+  @Mapping(target = "cdrVersionId", source = "dbWorkspace.cdrVersion")
+  @Mapping(target = "accessTierShortName", source = "dbWorkspace.cdrVersion.accessTier.shortName")
+  @Mapping(target = "permission", source = "accessLevel")
+  @Mapping(target = "cohort", source = "dbCohort")
+  // All workspaceResources have one object and all others are null.
+  @Mapping(target = "cohortReview", ignore = true)
+  @Mapping(target = "conceptSet", ignore = true)
+  @Mapping(target = "dataSet", ignore = true)
+  @Mapping(target = "notebook", ignore = true)
+  // This should be set when the resource is set
+  @Mapping(target = "modifiedTime", source = "dbCohort.lastModifiedTime")
+  WorkspaceResource dbWorkspaceAndDbCohortToWorkspaceResource(
+      DbWorkspace dbWorkspace, WorkspaceAccessLevel accessLevel, DbCohort dbCohort);
+
+  @Mapping(target = "workspaceId", source = "dbWorkspace.workspaceId")
+  @Mapping(target = "workspaceFirecloudName", source = "dbWorkspace.firecloudName")
+  @Mapping(target = "workspaceBillingStatus", source = "dbWorkspace.billingStatus")
+  @Mapping(target = "cdrVersionId", source = "dbWorkspace.cdrVersion")
+  @Mapping(target = "accessTierShortName", source = "dbWorkspace.cdrVersion.accessTier.shortName")
+  @Mapping(target = "permission", source = "accessLevel")
+  @Mapping(target = "cohortReview", source = "cohortReview")
+  // All workspaceResources have one object and all others are null.
+  @Mapping(target = "cohort", ignore = true)
+  @Mapping(target = "conceptSet", ignore = true)
+  @Mapping(target = "dataSet", ignore = true)
+  @Mapping(target = "notebook", ignore = true)
+  // This should be set when the resource is set
+  @Mapping(target = "modifiedTime", source = "cohortReview.lastModifiedTime")
+  WorkspaceResource dbWorkspaceAndCohortReviewToWorkspaceResource(
```

Comment:
This had **Db**CohortReview in the name, which was not correct.

(I hope to address the discrepancy between DB and Model objects here in a PR to follow soon)